idx/memory/tag_query.go

```diff
@@ -328,11 +318,19 @@ func (q *TagQuery) filterByMatch(resultSet TagIDs, byId map[string]*idx.Archive,
 					continue
```

Comment:
Should this just `break`? There shouldn't be another tag value for this tag key, right?

Reply:
hm github is not showing anything besides the `continue` line. are you talking about the case where we found an entry in `notMatchingTags` ?

Reply:
@shanson7 you're right. each def should only have one value per key

Reply:
i gave the inner loop a name for consistency with the outer one: https://github.com/grafana/metrictank/pull/729/commits/6fb73bfdd43b344d52267d256e9389db9a7b46ba